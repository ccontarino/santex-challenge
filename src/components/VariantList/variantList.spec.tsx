import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SelectComponent from './VariantList';

describe('SelectComponent', () => {
  const variants = [
    { id: '1', name: 'Variant 1' },
    { id: '2', name: 'Variant 2' },
    { id: '3', name: 'Variant 3' },
  ];
  const variantSelected = { id: '1', name: 'Variant 1' };
  const setVariant = jest.fn();

  it('should render the select component', () => {
    const { getByLabelText } = render(
      <SelectComponent
        variants={variants}
        variantSelected={variantSelected}
        setVariant={setVariant}
      />
    );
    const selectElement = getByLabelText('Select a variant');
    expect(selectElement).toBeInTheDocument();
  });

  it('should render the correct number of options', () => {
    const { getAllByRole } = render(
      <SelectComponent
        variants={variants}
        variantSelected={variantSelected}
        setVariant={setVariant}
      />
    );
    const optionElements = getAllByRole('option');
    expect(optionElements.length).toBe(variants.length);
  });

  it('should call setVariant when an option is selected', () => {
    const { getByLabelText } = render(
      <SelectComponent
        variants={variants}
        variantSelected={variantSelected}
        setVariant={setVariant}
      />
    );
    const selectElement = getByLabelText('Select a variant');
    fireEvent.change(selectElement, { target: { value: '2' } });
    expect(setVariant).toHaveBeenCalledWith(variants[1]);
  });
});