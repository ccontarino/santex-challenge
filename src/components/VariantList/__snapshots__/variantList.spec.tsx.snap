// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectComponent should render the select component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <label
          for="selectOption"
          style="display: block;"
        >
          Select a variant
        </label>
        <select
          class="sc-aXZVg bcxOaR"
          id="selectOption"
        >
          <option
            style="display: none;"
            value=""
          >
            Variant 1
          </option>
          <option
            value="1"
          >
            Variant 1
          </option>
          <option
            value="2"
          >
            Variant 2
          </option>
          <option
            value="3"
          >
            Variant 3
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <label
        for="selectOption"
        style="display: block;"
      >
        Select a variant
      </label>
      <select
        class="sc-aXZVg bcxOaR"
        id="selectOption"
      >
        <option
          style="display: none;"
          value=""
        >
          Variant 1
        </option>
        <option
          value="1"
        >
          Variant 1
        </option>
        <option
          value="2"
        >
          Variant 2
        </option>
        <option
          value="3"
        >
          Variant 3
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
